src/client/sky.h

```diff
@@ -111,17 +112,24 @@ class Sky : public scene::ISceneNode
 		const std::string &use_sun_tint);
 	void setInClouds(bool clouds) { m_in_clouds = clouds; }
 	void clearSkyboxTextures() { m_sky_params.textures.clear(); }
-	void addTextureToSkybox(const  std::string &texture, int material_id,
+	void addTextureToSkybox(const std::string &texture, int material_id,
 		ITextureSource *tsrc);
 	const video::SColorf &getCurrentStarColor() const { return m_star_color; }
+
+	// Note: the sky class doesn't even use these values. It just stores them.
+	// These should be moved to a better place in the future.
 	void setFogDistance(s16 fog_distance) { m_sky_params.fog_distance = fog_distance; }
 	s16 getFogDistance() const { return m_sky_params.fog_distance; }
 
 	void setFogStart(float fog_start) { m_sky_params.fog_start = fog_start; }
 	float getFogStart() const { return m_sky_params.fog_start; }
 
-	void setVolumetricLightStrength(float volumetric_light_strength) { m_sky_params.volumetric_light_strength = volumetric_light_strength; }
-	float getVolumetricLightStrength() const { return m_sky_params.volumetric_light_strength; }
+	void setFogColor(video::SColor v) { m_sky_params.fog_color = v; }
+	video::SColor getFogColor() const {
+		if (m_sky_params.fog_color.getAlpha() > 0)
```

Comment:
I assume you mean less than 255, because this isn't a float.
I don't know what happens if you set the fog to be semitransparent, does that make sense?

If you can imagine an usecase for overriding the fog color in a way that it's mixed with the normal fog color then I'll add this.

Reply:
Yeah sorry, 255.
I can't think of a usecase. I was just wondering because we do check for it being 0.